library-core/src/main/java/com/mikepenz/fastadapter/adapters/ModelAdapter.java

```diff
@@ -552,27 +474,39 @@ public void remapMappedTypes() {
      * @param itemCount the count of items which were removed
      */
     public ModelAdapter<Model, Item> removeRange(int position, int itemCount) {
-        //global position to relative
-        int length = mItems.size();
         int preItemCount = getFastAdapter().getPreItemCount(position);
-        //make sure we do not delete to many items
-        int saveItemCount = Math.min(itemCount, length - position + preItemCount);
-
-        for (int i = 0; i < saveItemCount; i++) {
-            mItems.remove(position - preItemCount);
-        }
 
-        getFastAdapter().notifyAdapterItemRangeRemoved(position, saveItemCount);
+        mItems.removeRange(position - preItemCount, itemCount);
         return this;
     }
 
     /**
      * removes all items of this adapter
      */
     public ModelAdapter<Model, Item> clear() {
-        int count = mItems.size();
         mItems.clear();
-        getFastAdapter().notifyAdapterItemRangeRemoved(getFastAdapter().getPreItemCountByOrder(getOrder()), count);
         return this;
     }
+
+    private final ListUpdateCallback listCallback = new ListUpdateCallback() {
```

Comment:
Moving it into *.fastadapter.utils.*DefaultListUpdateCallback would make it a bit cleaner.

Reply:
Technically, but there shouldn't be any other ones, so this should be an implementation detail of `ModelAdapter`.